module/zfs/zfs_sa.c

```diff
@@ -248,9 +248,21 @@ zfs_sa_set_xattr(znode_t *zp)
 	if (error) {
 		dmu_tx_abort(tx);
 	} else {
-		VERIFY0(sa_update(zp->z_sa_hdl, SA_ZPL_DXATTR(zfsvfs),
-		    obj, size, tx));
+		int count = 0;
+		sa_bulk_attr_t  bulk[2];
+		uint64_t ctime[2];
+
+		SA_ADD_BULK_ATTR(bulk, count, SA_ZPL_DXATTR(zfsvfs),
+		    NULL, obj, size);
+		/* Any xattr change is metadata change hence update the ctime */
+		zfs_tstamp_update_setup(zp, STATE_CHANGED, NULL, ctime);
```

Comment:
style: it might be nice to move the `zfs_tstamp_update_setup()` call easier so all the `SA_ADD_BULK_ATTR()` calls are together.

Reply:
done